### PR TITLE
Merge from origin ua-parser-js

### PR DESCRIPTION
**Pull request by dnshi**
# Summary

This PR is to merge ua-parser-js into this repo and fix missing device types.

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ua-parser-js",
-  "version": "0.7.28",
+  "version": "0.7.31",
```
Comment (qingzhuozhen) [inline bower.json:3]: why `0.7.31`?

**Review by qingzhuozhen — COMMENTED** [review]
What's the UA-parser version we are syncing with?

File: bower.json (hunk 1 above)
Comment (dnshi) [reply bower.json:3]: The latest version from ua parser is `0.7.31`, so when I pull the code and merge it, it becomes this version number.

**Comment by dnshi** [comment]
@qingzhuozhen  I pulled the latest version, which is `0.7.31`.

**Review by qingzhuozhen — APPROVED** [review]

**Review by yuhao900914 — APPROVED** [review]

**Pull request merged**